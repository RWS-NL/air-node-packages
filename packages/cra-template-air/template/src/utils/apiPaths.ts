export enum API_METHOD {
  GET = 'get',
  POST = 'post',
  PUT = 'put',
  DELETE = 'delete'
}

export enum ENDPOINTS {
  CONFIG = 'config'
}

export enum API_VERSION {
  v1 = 'v1'
}

export enum ACTION_ENDPOINTS {
  GET_CONFIG = 'getConfig'
}

export const CONFIG_PATH = (version: API_VERSION) => `/api/${version}/${ENDPOINTS.CONFIG}`;
