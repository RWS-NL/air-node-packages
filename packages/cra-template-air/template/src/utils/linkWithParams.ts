import { parse, stringify } from '@favware/querystring';
import windowLocationService from './WindowLocationService';

export const linkWithParams = <T>(
  path: string,
  paramsToAdd: T,
  shouldMerge?: boolean,
  search?: string
) => {
  if (!search) search = windowLocationService.search;

  const currentParams = parse(search);
  let newSearchString = stringify(paramsToAdd);

  if (shouldMerge) {
    newSearchString = stringify({ ...currentParams, ...paramsToAdd });
  }

  if (newSearchString !== '') {
    return `${path}?${newSearchString}`;
  }

  return path;
};

export default linkWithParams;
