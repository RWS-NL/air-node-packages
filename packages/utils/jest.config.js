module.exports = {
  projects: [
    {
      displayName: 'ts-jest',
      testMatch: ['<rootDir>/__tests__/**/*.ts'],
      preset: 'ts-jest',
      testEnvironment: 'node',
      testRunner: 'jest-circus/runner'
    },
    {
      displayName: 'react-test',
      testMatch: ['<rootDir>/__tests__/**/*.tsx'],
      snapshotSerializers: ['enzyme-to-json/serializer'],
      setupFiles: ['<rootDir>/jest.setup.ts']
    }
  ]
};
