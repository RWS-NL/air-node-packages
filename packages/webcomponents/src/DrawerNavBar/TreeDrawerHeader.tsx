import Divider from '@material-ui/core/Divider';
import FormControl from '@material-ui/core/FormControl';
import Grid from '@material-ui/core/Grid';
import IconButton from '@material-ui/core/IconButton';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import MenuItem from '@material-ui/core/MenuItem';
import Select from '@material-ui/core/Select';
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
import ArrowBackIcon from '@material-ui/icons/ArrowBack';
import ArrowForward from '@material-ui/icons/ArrowForward';
import InfoIcon from '@material-ui/icons/Info';
import clsx from 'clsx';
import React, { memo } from 'react';
import { Else, If, Then, When } from 'react-if';
import { Tooltip } from '../Tooltip';
import { TreeDrawerProps } from './DrawerProps';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    treeDropdownGrid: {
      maxHeight: '1rem',
      marginTop: '-2.5rem'
    },
    treeSelector: {
      margin: theme.spacing(1),
      minWidth: 120
    },
    treeDrawerCloseIcon: {
      marginRight: theme.spacing(1)
    },
    smallInfoButton: {
      width: '0.6em',
      height: '0.6em'
    },
    treeInfoButton: {
      paddingTop: 0,
      paddingBottom: 6
    },
    closeListItem: {
      justifyContent: 'flex-end'
    },
    divider: {
      backgroundColor: '#CCE1EB' // Dark Blue 20
    }
  })
);

export const TreeDrawerHeader = memo((treeDrawer: TreeDrawerProps) => {
  const classes = useStyles();

  return (
    <>
      <List>
        <ListItem
          button
          disableGutters={treeDrawer.open}
          onClick={treeDrawer.toggleDrawer}
          classes={{
            root: clsx({ [classes.closeListItem]: treeDrawer.open })
          }}
        >
          <Grid
            container
            direction='row'
            justify='space-between'
            alignItems='center'
            alignContent='stretch'
            wrap='nowrap'
            spacing={0}
          >
            <When condition={treeDrawer.open}>
              <Grid item xs={1} />
              <Grid item xs={8} classes={{ root: classes.treeDropdownGrid }}>
                <ListItemText
                  disableTypography
                  primary={
                    <FormControl className={classes.treeSelector} size='small'>
                      <Select
                        value={treeDrawer.currentDropdownValue}
                        onChange={(event) => {
                          event.preventDefault();
                          treeDrawer.hanbdleDropdownChange(event);
                          event.stopPropagation();
                        }}
                        onClose={(event) => {
                          event.preventDefault();
                          event.stopPropagation();
                        }}
                        onOpen={(event) => {
                          event.preventDefault();
                          event.stopPropagation();
                        }}
                        fullWidth
                        displayEmpty
                        disableUnderline
                        MenuProps={{
                          getContentAnchorEl: null,
                          anchorOrigin: {
                            vertical: 'bottom',
                            horizontal: 'center'
                          },
                          transformOrigin: {
                            vertical: 'top',
                            horizontal: 'center'
                          },
                          elevation: 2
                        }}
                        renderValue={(value) => <b>{value as string}</b>}
                      >
                        {treeDrawer.dropdownValues.map(({ value, label }, index) => (
                          <MenuItem key={index} value={value}>
                            {label}
                          </MenuItem>
                        ))}
                      </Select>
                    </FormControl>
                  }
                />
              </Grid>
              <Grid item xs={1}>
                <Tooltip title={treeDrawer.tooltipText}>
                  <IconButton color='primary' size='small' classes={{ root: classes.treeInfoButton }}>
                    <InfoIcon className={classes.smallInfoButton} />
                  </IconButton>
                </Tooltip>
              </Grid>
            </When>
            <Grid item xs={1} classes={{ root: classes.treeDrawerCloseIcon }}>
              <ListItemIcon>
                <If condition={treeDrawer.open}>
                  <Then>
                    <ArrowBackIcon color='primary' />
                  </Then>
                  <Else>
                    <ArrowForward color='primary' />
                  </Else>
                </If>
              </ListItemIcon>
            </Grid>
          </Grid>
        </ListItem>
      </List>
      <Divider classes={{ root: classes.divider }} />
    </>
  );
});
