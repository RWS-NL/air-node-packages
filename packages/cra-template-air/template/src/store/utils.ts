import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import { ACTION_ENDPOINTS, API_METHOD } from 'utils/apiPaths';
import ApiClient, { FetchResponse, FetchResultTypes } from 'utils/apiClient';
import { call, put } from 'redux-saga/effects';
import { AIRHandledError, HTTP_SUCCESS_CODES, defaultToastrOptions } from 'utils/constants';
import { BAD_REQUEST, FORBIDDEN, NOT_FOUND, REQUEST_TIMEOUT } from 'http-status';
import { actions as toastrActions, toastType } from 'react-redux-toastr';
import i18n from 'config/i18n';

export const composeEnhancers = composeWithDevTools({});

/**
 * Gets some data from the backend
 * @param path The API path to get data from
 * @returns the data from the backend in JSON format
 */
export function* getJson<T>(path: string) {
  const response: FetchResponse<T> = yield call([ApiClient, 'fetch'], API_METHOD.GET, path, FetchResultTypes.JSON);

  if (!response) throw new AIRHandledError(REQUEST_TIMEOUT, 'timed_out_no_response');

  if (!Object.keys(response.data).length && !HTTP_SUCCESS_CODES.includes(response.status)) {
    throw new AIRHandledError(response.status || BAD_REQUEST, 'bad_request');
  }

  return response.data;
}

/**
 * Posts some data to an endpoint on the backend
 * @param path The API path to post data to
 * @param body The object to include as the body of the post
 * @returns The body returned by the backend in JSON format
 */
export function* postJson<T>(path: string, body: any) {
  const response: FetchResponse<T> = yield call(
    [ApiClient, 'fetch'],
    API_METHOD.POST,
    path,
    FetchResultTypes.JSON,
    body
  );

  if (!response) throw new AIRHandledError(REQUEST_TIMEOUT, 'timed_out_no_response');

  if (!Object.keys(response.data).length && !HTTP_SUCCESS_CODES.includes(response.status)) {
    throw new AIRHandledError(response.status || BAD_REQUEST, 'bad_request');
  }

  return response.data;
}

/**
 * Puts some data to an endpoint on the backend
 * @param path The API path to put data to
 * @param body The object to include as the body of the put
 * @returns The body returned by the backend in JSON format
 */
export function* putJson<T>(path: string, body: any) {
  const response: FetchResponse<T> = yield call(
    [ApiClient, 'fetch'],
    API_METHOD.PUT,
    path,
    FetchResultTypes.JSON,
    body
  );

  if (!response) throw new AIRHandledError(REQUEST_TIMEOUT, 'timed_out_no_response');

  if (response.data && !Object.keys(response.data).length && !HTTP_SUCCESS_CODES.includes(response.status)) {
    throw new AIRHandledError(response.status || BAD_REQUEST, 'bad_request');
  }

  return response.data;
}

/**
 * Generates a toastr message
 * @param message The message to show in the toastr
 * @param type Optional type of the toastr
 */
export const displayToastr = (message: string, type: toastType = 'success') => {
  return toastrActions.add({
    ...defaultToastrOptions,
    message,
    type
  });
};

/**
 * Handles HTTP status codes by generating toastrs
 * @param code The HTTP status code that was returned
 * @param action The endpoint where the status code occured on
 */
export function* errorHandler(code: number, action: ACTION_ENDPOINTS) {
  switch (code) {
    case BAD_REQUEST:
      return yield put(displayToastr(i18n.t(`toasts.badRequest.${action}`), 'error'));
    case FORBIDDEN:
      return yield put(displayToastr(i18n.t(`toasts.forbidden.${action}`), 'error'));
    case NOT_FOUND:
      return yield put(displayToastr(i18n.t(`toasts.notFound.${action}`), 'error'));
    case REQUEST_TIMEOUT:
      return yield put(displayToastr(i18n.t(`toasts.requestTimeout.${action}`), 'error'));
    default:
      return yield put(displayToastr(i18n.t('toasts.unhandledError'), 'error'));
  }
}
