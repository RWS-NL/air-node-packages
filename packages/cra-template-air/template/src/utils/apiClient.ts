import { BAD_REQUEST, FORBIDDEN, INTERNAL_SERVER_ERROR, NOT_FOUND, UNAUTHORIZED } from 'http-status';
import { API_METHOD } from './apiPaths';
import { AIRHandledError } from './constants';

export enum FetchResultTypes {
  JSON,
  Blob,
  Text,
  Result
}

export interface FetchResponse<T> {
  data: T;
  status: number;
  statusText: string;
  headers: Headers;
}

export default class ApiClient {
  public static accessToken: string | undefined;
  private static apiUrl: string | undefined;

  public get accessToken(): string | undefined {
    return ApiClient.accessToken;
  }

  public set accessToken(value: string | undefined) {
    ApiClient.accessToken = value;
  }

  public get apiUrl(): string | undefined {
    return ApiClient.apiUrl;
  }

  public set apiUrl(value: string | undefined) {
    ApiClient.apiUrl = value;
  }

  public constructor(apiUrl?: string) {
    this.apiUrl = apiUrl || '';
  }

  public static async fetch(method: API_METHOD, url: string, type: FetchResultTypes.Blob, body?: any): Promise<FetchResponse<Blob>>;
  public static async fetch(method: API_METHOD, url: string, type: FetchResultTypes.Text, body?: any): Promise<FetchResponse<string>>;
  public static async fetch(method: API_METHOD, url: string, type: FetchResultTypes.Result, body?: any): Promise<Response>;
  public static async fetch<T extends unknown>(method: API_METHOD, url: string, type: FetchResultTypes.JSON, body?: any): Promise<FetchResponse<T>>;
  public static async fetch<T extends unknown>(method: API_METHOD, url: string, type: FetchResultTypes, body?: any): Promise<Response | FetchResponse<Blob> | FetchResponse<string> | FetchResponse<T>>;
  public static async fetch<T extends unknown>(method: API_METHOD, url: string, type: FetchResultTypes, body?: any) {
    try {
      const response: Response = await fetch(`${ApiClient.apiUrl || ''}${url}`, {
        method,
        headers: {
          'Content-Type': 'application/json',
          Authorization: `Bearer ${ApiClient.accessToken}`
        },
        body: JSON.stringify(body)
      });

      if (!response.ok) throw new AIRHandledError(response.status, `${response.status} error`);

      switch (type) {
        case FetchResultTypes.Result:
          return response;
        case FetchResultTypes.Blob:
          return {
            data: await response.blob(),
            status: response.status,
            statusText: response.statusText,
            headers: response.headers
          } as FetchResponse<Blob>;
        case FetchResultTypes.JSON:
          return {
            data: await ApiClient.parseBody<T>(response),
            status: response.status,
            statusText: response.statusText,
            headers: response.headers
          } as FetchResponse<T>;
        case FetchResultTypes.Text:
          return {
            data: await response.text(),
            status: response.status,
            statusText: response.statusText,
            headers: response.headers
          } as FetchResponse<string>;
        default:
          throw new TypeError(`Unknown fetch type ${type}`);
      }
    } catch (err) {
      if (err.toString().includes(BAD_REQUEST.toString())) throw new AIRHandledError(BAD_REQUEST, '400 error');
      if (err.toString().includes(UNAUTHORIZED.toString())) throw new AIRHandledError(UNAUTHORIZED, '401 error');
      if (err.toString().includes(FORBIDDEN.toString())) throw new AIRHandledError(FORBIDDEN, '403 error');
      if (err.toString().includes(NOT_FOUND.toString())) throw new AIRHandledError(NOT_FOUND, '404 error');

      throw new AIRHandledError(INTERNAL_SERVER_ERROR, 'Internal Fetch Error');
    }
  }

  private static async parseBody<T>(response: Response): Promise<T | undefined> {
    const contentType = response.headers.get('content-type');

    if (contentType && contentType.includes('application/json')) {
      return (await response.json()) as Promise<T>;
    }

    return undefined;
  }

  public async fetch(method: API_METHOD, url: string, type: FetchResultTypes.Blob, body?: any): Promise<FetchResponse<Blob>>;
  public async fetch(method: API_METHOD, url: string, type: FetchResultTypes.Text, body?: any): Promise<FetchResponse<string>>;
  public async fetch(method: API_METHOD, url: string, type: FetchResultTypes.Result, body?: any): Promise<Response>;
  public async fetch<T extends unknown>(method: API_METHOD, url: string, type: FetchResultTypes.JSON, body?: any): Promise<FetchResponse<T>>;
  public async fetch<T extends unknown>(method: API_METHOD, url: string, type: FetchResultTypes, body?: any): Promise<Response | FetchResponse<Blob> | FetchResponse<string> | FetchResponse<T>>;
  public async fetch<T extends unknown>(method: API_METHOD, url: string, type: FetchResultTypes, body?: any) {
    return ApiClient.fetch<T>(method, url, type, body);
  }
}
