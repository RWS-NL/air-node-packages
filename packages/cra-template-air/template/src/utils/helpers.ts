import { capitilizeFirstLetter, snakeToCamelCase } from '@rws-air/utils';
import { Oauth2Permissions, UserPrivileges } from 'store/oauth2/oauth2Types';

export const noOp = () => undefined;

export const transformPrivileges = (privileges: UserPrivileges) =>
  privileges
    .map((entry) => `can${capitilizeFirstLetter(snakeToCamelCase(entry.toLowerCase()))}`)
    .reduce<Oauth2Permissions>((prevVal, curVal) => ({ [curVal]: true, ...prevVal }), {});
