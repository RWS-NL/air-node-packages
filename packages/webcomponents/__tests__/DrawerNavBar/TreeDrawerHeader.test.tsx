import { TreeDrawerHeader, TreeDrawerHeaderDropdown } from '@DrawerNavBar/TreeDrawerHeader';
import { MenuItem, Select } from '@material-ui/core';
import { mockHandleBoomChange, treeDrawerProps } from '@Resource/commonDrawerNavBarProps';
import { createEvent, toBeCalled } from '@rws-air/utils';
import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

describe('TreeDrawerHeader', () => {
  describe('Component Testing', () => {
    let treeDrawerHeaderDropdown: ShallowWrapper;

    const mockStopPropagation = jest.fn();
    const mockPreventDefault = jest.fn();

    const dummyEvent = createEvent('TWO', { stopPropagation: mockStopPropagation, preventDefault: mockPreventDefault });

    beforeAll(() => {
      treeDrawerHeaderDropdown = shallow(<TreeDrawerHeaderDropdown {...treeDrawerProps} />);
    });
    test('GIVEN change on Select THEN triggers change from props', () => {
      const selectMenu = treeDrawerHeaderDropdown.find(Select);

      selectMenu.simulate('change', dummyEvent);

      toBeCalled(mockHandleBoomChange, 1, dummyEvent);
    });

    test('GIVEN props THEN renders MenuItems', () => {
      const menuItems = treeDrawerHeaderDropdown.find(MenuItem);

      expect(menuItems).toHaveLength(treeDrawerProps.dropdownValues.length);
    });
  });

  describe('Snapshot Testing', () => {
    test('Required Props', () => {
      const treeDrawerHeader = shallow(<TreeDrawerHeader {...treeDrawerProps} />);
      expect(treeDrawerHeader).toMatchSnapshot();
    });
  });
});
