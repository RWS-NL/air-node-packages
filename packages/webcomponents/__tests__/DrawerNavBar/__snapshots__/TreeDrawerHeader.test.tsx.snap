// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TreeDrawerHeader Snapshot Testing Required Props 1`] = `
<Fragment>
  <WithStyles(ForwardRef(List))>
    <WithStyles(ForwardRef(ListItem))
      button={true}
      classes={
        Object {
          "root": "makeStyles-closeListItem-6",
        }
      }
      disableGutters={true}
      onClick={[MockFunction]}
    >
      <WithStyles(ForwardRef(Grid))
        alignContent="stretch"
        alignItems="center"
        container={true}
        direction="row"
        justify="space-between"
        spacing={0}
        wrap="nowrap"
      >
        <When
          condition={true}
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={1}
          />
          <WithStyles(ForwardRef(Grid))
            classes={
              Object {
                "root": "makeStyles-treeDropdownGrid-1",
              }
            }
            item={true}
            xs={8}
          >
            <WithStyles(ForwardRef(ListItemText))
              disableTypography={true}
              primary={
                <Memo
                  content={
                    <WithStyles(ForwardRef(Paper))
                      elevation={2}
                    >
                      <WithStyles(ForwardRef(List))>
                        <WithStyles(ForwardRef(ListItem))
                          button={true}
                        >
                          <WithStyles(ForwardRef(ListItemIcon))>
                            <Memo(MoveToInboxIcon) />
                          </WithStyles(ForwardRef(ListItemIcon))>
                          <WithStyles(ForwardRef(ListItemText))
                            primary="Inbox"
                            primaryTypographyProps={
                              Object {
                                "color": "primary",
                                "variant": "body2",
                              }
                            }
                          />
                        </WithStyles(ForwardRef(ListItem))>
                        <WithStyles(ForwardRef(ListItem))
                          button={true}
                        >
                          <WithStyles(ForwardRef(ListItemIcon))>
                            <Memo(DraftsIcon) />
                          </WithStyles(ForwardRef(ListItemIcon))>
                          <WithStyles(ForwardRef(ListItemText))
                            primary="Drafts"
                            primaryTypographyProps={
                              Object {
                                "color": "primary",
                                "variant": "body2",
                              }
                            }
                          />
                        </WithStyles(ForwardRef(ListItem))>
                        <WithStyles(ForwardRef(ListItem))
                          button={true}
                        >
                          <WithStyles(ForwardRef(ListItemIcon))>
                            <Memo(MoveToInboxIcon) />
                          </WithStyles(ForwardRef(ListItemIcon))>
                          <WithStyles(ForwardRef(ListItemText))
                            primary="Inbox"
                            primaryTypographyProps={
                              Object {
                                "color": "primary",
                                "variant": "body2",
                              }
                            }
                          />
                        </WithStyles(ForwardRef(ListItem))>
                        <WithStyles(ForwardRef(ListItem))
                          button={true}
                        >
                          <WithStyles(ForwardRef(ListItemIcon))>
                            <Memo(DraftsIcon) />
                          </WithStyles(ForwardRef(ListItemIcon))>
                          <WithStyles(ForwardRef(ListItemText))
                            primary="Drafts"
                            primaryTypographyProps={
                              Object {
                                "color": "primary",
                                "variant": "body2",
                              }
                            }
                          />
                        </WithStyles(ForwardRef(ListItem))>
                        <WithStyles(ForwardRef(ListItem))
                          button={true}
                        >
                          <WithStyles(ForwardRef(ListItemIcon))>
                            <Memo(MoveToInboxIcon) />
                          </WithStyles(ForwardRef(ListItemIcon))>
                          <WithStyles(ForwardRef(ListItemText))
                            primary="Inbox"
                            primaryTypographyProps={
                              Object {
                                "color": "primary",
                                "variant": "body2",
                              }
                            }
                          />
                        </WithStyles(ForwardRef(ListItem))>
                        <WithStyles(ForwardRef(ListItem))
                          button={true}
                        >
                          <WithStyles(ForwardRef(ListItemIcon))>
                            <Memo(DraftsIcon) />
                          </WithStyles(ForwardRef(ListItemIcon))>
                          <WithStyles(ForwardRef(ListItemText))
                            primary="Drafts"
                            primaryTypographyProps={
                              Object {
                                "color": "primary",
                                "variant": "body2",
                              }
                            }
                          />
                        </WithStyles(ForwardRef(ListItem))>
                        <WithStyles(ForwardRef(ListItem))
                          button={true}
                        >
                          <WithStyles(ForwardRef(ListItemIcon))>
                            <Memo(MoveToInboxIcon) />
                          </WithStyles(ForwardRef(ListItemIcon))>
                          <WithStyles(ForwardRef(ListItemText))
                            primary="Inbox"
                            primaryTypographyProps={
                              Object {
                                "color": "primary",
                                "variant": "body2",
                              }
                            }
                          />
                        </WithStyles(ForwardRef(ListItem))>
                        <WithStyles(ForwardRef(ListItem))
                          button={true}
                        >
                          <WithStyles(ForwardRef(ListItemIcon))>
                            <Memo(DraftsIcon) />
                          </WithStyles(ForwardRef(ListItemIcon))>
                          <WithStyles(ForwardRef(ListItemText))
                            primary="Drafts"
                            primaryTypographyProps={
                              Object {
                                "color": "primary",
                                "variant": "body2",
                              }
                            }
                          />
                        </WithStyles(ForwardRef(ListItem))>
                        <WithStyles(ForwardRef(ListItem))
                          button={true}
                        >
                          <WithStyles(ForwardRef(ListItemIcon))>
                            <Memo(MoveToInboxIcon) />
                          </WithStyles(ForwardRef(ListItemIcon))>
                          <WithStyles(ForwardRef(ListItemText))
                            primary="Inbox"
                            primaryTypographyProps={
                              Object {
                                "color": "primary",
                                "variant": "body2",
                              }
                            }
                          />
                        </WithStyles(ForwardRef(ListItem))>
                        <WithStyles(ForwardRef(ListItem))
                          button={true}
                        >
                          <WithStyles(ForwardRef(ListItemIcon))>
                            <Memo(DraftsIcon) />
                          </WithStyles(ForwardRef(ListItemIcon))>
                          <WithStyles(ForwardRef(ListItemText))
                            primary="Drafts"
                            primaryTypographyProps={
                              Object {
                                "color": "primary",
                                "variant": "body2",
                              }
                            }
                          />
                        </WithStyles(ForwardRef(ListItem))>
                        <WithStyles(ForwardRef(ListItem))
                          button={true}
                        >
                          <WithStyles(ForwardRef(ListItemIcon))>
                            <Memo(MoveToInboxIcon) />
                          </WithStyles(ForwardRef(ListItemIcon))>
                          <WithStyles(ForwardRef(ListItemText))
                            primary="Inbox"
                            primaryTypographyProps={
                              Object {
                                "color": "primary",
                                "variant": "body2",
                              }
                            }
                          />
                        </WithStyles(ForwardRef(ListItem))>
                        <WithStyles(ForwardRef(ListItem))
                          button={true}
                        >
                          <WithStyles(ForwardRef(ListItemIcon))>
                            <Memo(DraftsIcon) />
                          </WithStyles(ForwardRef(ListItemIcon))>
                          <WithStyles(ForwardRef(ListItemText))
                            primary="Drafts"
                            primaryTypographyProps={
                              Object {
                                "color": "primary",
                                "variant": "body2",
                              }
                            }
                          />
                        </WithStyles(ForwardRef(ListItem))>
                        <WithStyles(ForwardRef(ListItem))
                          button={true}
                        >
                          <WithStyles(ForwardRef(ListItemIcon))>
                            <Memo(MoveToInboxIcon) />
                          </WithStyles(ForwardRef(ListItemIcon))>
                          <WithStyles(ForwardRef(ListItemText))
                            primary="Inbox"
                            primaryTypographyProps={
                              Object {
                                "color": "primary",
                                "variant": "body2",
                              }
                            }
                          />
                        </WithStyles(ForwardRef(ListItem))>
                        <WithStyles(ForwardRef(ListItem))
                          button={true}
                        >
                          <WithStyles(ForwardRef(ListItemIcon))>
                            <Memo(DraftsIcon) />
                          </WithStyles(ForwardRef(ListItemIcon))>
                          <WithStyles(ForwardRef(ListItemText))
                            primary="Drafts"
                            primaryTypographyProps={
                              Object {
                                "color": "primary",
                                "variant": "body2",
                              }
                            }
                          />
                        </WithStyles(ForwardRef(ListItem))>
                        <WithStyles(ForwardRef(ListItem))
                          button={true}
                        >
                          <WithStyles(ForwardRef(ListItemIcon))>
                            <Memo(MoveToInboxIcon) />
                          </WithStyles(ForwardRef(ListItemIcon))>
                          <WithStyles(ForwardRef(ListItemText))
                            primary="Inbox"
                            primaryTypographyProps={
                              Object {
                                "color": "primary",
                                "variant": "body2",
                              }
                            }
                          />
                        </WithStyles(ForwardRef(ListItem))>
                        <WithStyles(ForwardRef(ListItem))
                          button={true}
                        >
                          <WithStyles(ForwardRef(ListItemIcon))>
                            <Memo(DraftsIcon) />
                          </WithStyles(ForwardRef(ListItemIcon))>
                          <WithStyles(ForwardRef(ListItemText))
                            primary="Drafts"
                            primaryTypographyProps={
                              Object {
                                "color": "primary",
                                "variant": "body2",
                              }
                            }
                          />
                        </WithStyles(ForwardRef(ListItem))>
                        <WithStyles(ForwardRef(ListItem))
                          button={true}
                        >
                          <WithStyles(ForwardRef(ListItemIcon))>
                            <Memo(MoveToInboxIcon) />
                          </WithStyles(ForwardRef(ListItemIcon))>
                          <WithStyles(ForwardRef(ListItemText))
                            primary="Inbox"
                            primaryTypographyProps={
                              Object {
                                "color": "primary",
                                "variant": "body2",
                              }
                            }
                          />
                        </WithStyles(ForwardRef(ListItem))>
                        <WithStyles(ForwardRef(ListItem))
                          button={true}
                        >
                          <WithStyles(ForwardRef(ListItemIcon))>
                            <Memo(DraftsIcon) />
                          </WithStyles(ForwardRef(ListItemIcon))>
                          <WithStyles(ForwardRef(ListItemText))
                            primary="Drafts"
                            primaryTypographyProps={
                              Object {
                                "color": "primary",
                                "variant": "body2",
                              }
                            }
                          />
                        </WithStyles(ForwardRef(ListItem))>
                        <WithStyles(ForwardRef(ListItem))
                          button={true}
                        >
                          <WithStyles(ForwardRef(ListItemIcon))>
                            <Memo(MoveToInboxIcon) />
                          </WithStyles(ForwardRef(ListItemIcon))>
                          <WithStyles(ForwardRef(ListItemText))
                            primary="Inbox"
                            primaryTypographyProps={
                              Object {
                                "color": "primary",
                                "variant": "body2",
                              }
                            }
                          />
                        </WithStyles(ForwardRef(ListItem))>
                        <WithStyles(ForwardRef(ListItem))
                          button={true}
                        >
                          <WithStyles(ForwardRef(ListItemIcon))>
                            <Memo(DraftsIcon) />
                          </WithStyles(ForwardRef(ListItemIcon))>
                          <WithStyles(ForwardRef(ListItemText))
                            primary="Drafts"
                            primaryTypographyProps={
                              Object {
                                "color": "primary",
                                "variant": "body2",
                              }
                            }
                          />
                        </WithStyles(ForwardRef(ListItem))>
                        <WithStyles(ForwardRef(ListItem))
                          button={true}
                        >
                          <WithStyles(ForwardRef(ListItemIcon))>
                            <Memo(MoveToInboxIcon) />
                          </WithStyles(ForwardRef(ListItemIcon))>
                          <WithStyles(ForwardRef(ListItemText))
                            primary="Inbox"
                            primaryTypographyProps={
                              Object {
                                "color": "primary",
                                "variant": "body2",
                              }
                            }
                          />
                        </WithStyles(ForwardRef(ListItem))>
                      </WithStyles(ForwardRef(List))>
                    </WithStyles(ForwardRef(Paper))>
                  }
                  currentDropdownValue="THREE"
                  dropdownValues={
                    Array [
                      Object {
                        "label": "ONE",
                        "value": "ONE",
                      },
                      Object {
                        "label": "TWO",
                        "value": "TWO",
                      },
                      Object {
                        "label": "THREE",
                        "value": "THREE",
                      },
                      Object {
                        "label": "FOUR",
                        "value": "FOUR",
                      },
                      Object {
                        "label": "FIVE",
                        "value": "FIVE",
                      },
                      Object {
                        "label": "SIX",
                        "value": "SIX",
                      },
                    ]
                  }
                  hanbdleDropdownChange={
                    [MockFunction] {
                      "calls": Array [
                        Array [
                          Object {
                            "preventDefault": [MockFunction] {
                              "calls": Array [
                                Array [],
                              ],
                              "results": Array [
                                Object {
                                  "type": "return",
                                  "value": undefined,
                                },
                              ],
                            },
                            "stopPropagation": [MockFunction] {
                              "calls": Array [
                                Array [],
                              ],
                              "results": Array [
                                Object {
                                  "type": "return",
                                  "value": undefined,
                                },
                              ],
                            },
                            "target": Object {
                              "value": "TWO",
                            },
                          },
                        ],
                      ],
                      "results": Array [
                        Object {
                          "type": "return",
                          "value": undefined,
                        },
                      ],
                    }
                  }
                  minWidth={400}
                  onBorderDrag={[MockFunction]}
                  open={true}
                  toggleDrawer={[MockFunction]}
                  tooltipText="Hier kunt u andere type bomen selecteren. Druk op het menu links van deze knop en selecteer dan uw gewenste boom type."
                  width={240}
                />
              }
            />
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={1}
          >
            <Memo()
              title="Hier kunt u andere type bomen selecteren. Druk op het menu links van deze knop en selecteer dan uw gewenste boom type."
            >
              <WithStyles(ForwardRef(IconButton))
                classes={
                  Object {
                    "root": "makeStyles-treeInfoButton-5",
                  }
                }
                color="primary"
                size="small"
              >
                <InfoIcon
                  className="makeStyles-smallInfoButton-4"
                />
              </WithStyles(ForwardRef(IconButton))>
            </Memo()>
          </WithStyles(ForwardRef(Grid))>
        </When>
        <WithStyles(ForwardRef(Grid))
          classes={
            Object {
              "root": "makeStyles-treeDrawerCloseIcon-3",
            }
          }
          item={true}
          xs={1}
        >
          <WithStyles(ForwardRef(ListItemIcon))>
            <If
              condition={true}
            >
              <Then>
                <ArrowBackIcon
                  color="primary"
                />
              </Then>
              <Else>
                <ArrowForwardIcon
                  color="primary"
                />
              </Else>
            </If>
          </WithStyles(ForwardRef(ListItemIcon))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(ListItem))>
  </WithStyles(ForwardRef(List))>
  <WithStyles(ForwardRef(Divider))
    classes={
      Object {
        "root": "makeStyles-divider-7",
      }
    }
  />
</Fragment>
`;
